src/http_server_handle_req.c

```diff
@@ -52,6 +52,111 @@ http_server_gen_resp_status_json(const json_network_info_t *const p_info, void *
+            return http_server_resp_503();
+        }
+        LOG_INFO("ap.json: %s", ((NULL != p_buff) ? p_buff : "NULL"));
+        const http_server_resp_t resp = http_server_resp_200_json(p_buff);
+        return resp;
+    }
+    if (0 == strcmp(p_file_name, "status.json"))
+    {
+        http_server_update_last_http_status_request();
+
+        http_server_resp_t                       http_resp = { 0 };
+        http_server_gen_resp_status_json_param_t params    = {
+            .p_http_resp          = &http_resp,
+            .flag_access_from_lan = flag_access_from_lan,
+        };
+        const os_delta_ticks_t ticks_to_wait = 10U;
+        json_network_info_do_const_action_with_timeout(&http_server_gen_resp_status_json, &params, ticks_to_wait);
+        return http_resp;
+    }
+    if (0 == strcmp(p_file_name, "auth.html"))
+    {
+        return wifi_manager_cb_on_http_get(p_file_name, flag_access_from_lan, p_resp_auth_check);
+    }
+    return http_server_resp_404();
+}
+
+static http_server_resp_t
+http_server_handle_req_get_path_without_extension_api_key_not_used(
+    const char *const                    p_file_name,
+    const http_server_auth_info_t *const p_auth_info,
+    const http_server_resp_t *const      p_resp_auth_check,
+    http_header_extra_fields_t *const    p_extra_header_fields)
+{
+    if (HTTP_RESP_CODE_200 != p_resp_auth_check->http_resp_code)
+    {
+        if (HTTP_SERVER_AUTH_TYPE_RUUVI == p_auth_info->auth_type)
+        {
+            snprintf(
+                p_extra_header_fields->buf,
+                sizeof(p_extra_header_fields->buf),
+                "Set-Cookie: %s=/%s",
+                HTTP_SERVER_AUTH_RUUVI_COOKIE_PREV_URL,
+                p_file_name);
+            return http_server_resp_302();
+        }
+        return *p_resp_auth_check;
+    }
+    return http_server_resp_data_in_flash(HTTP_CONENT_TYPE_TEXT_PLAIN, NULL, 0, HTTP_CONENT_ENCODING_NONE, NULL);
+}
+
+static http_server_resp_t
+http_server_handle_req_get_path_without_extension(
+    const char *const                    p_file_name,
+    const http_server_auth_info_t *const p_auth_info,
+    const wifi_ssid_t *const             p_ap_ssid,
+    const http_server_resp_t *const      p_resp_auth_check,
+    const http_server_auth_api_key_e     access_by_api_key,
+    http_header_extra_fields_t *const    p_extra_header_fields)
+{
+    switch (access_by_api_key)
+    {
+        case HTTP_SERVER_AUTH_API_KEY_NOT_USED:
+            return http_server_handle_req_get_path_without_extension_api_key_not_used(
+                p_file_name,
+                p_auth_info,
+                p_resp_auth_check,
+                p_extra_header_fields);
+        case HTTP_SERVER_AUTH_API_KEY_ALLOWED:
+            break;
+        case HTTP_SERVER_AUTH_API_KEY_PROHIBITED:
+            return http_server_resp_401_json(http_server_fill_auth_json_bearer_failed(p_ap_ssid));
+    }
```

Comment:
> Please add a default statement to this switch-case

I think that it's not necessary, because gcc can detect such problems.
For example, if I add new item 'HTTP_SERVER_AUTH_API_KEY_UNUSED' to this enum, then compiler will generate an error:

```
/home/alexey/dev/ruuvi.gateway_esp.c/components/esp32-wifi-manager/src/http_server_handle_req.c: In function 'http_server_handle_req_get_path_without_extension':
/home/alexey/dev/ruuvi.gateway_esp.c/components/esp32-wifi-manager/src/http_server_handle_req.c:144:5: error: enumeration value 'HTTP_SERVER_AUTH_API_KEY_UNUSED' not handled in switch [-Werror=switch]
     switch (access_by_api_key)
     ^~~~~~
```

On the other hand, adding 'default' section makes it harder to find all the places that need to be updated after adding a new item to enum.

So, do you think the protection provided by the compiler will be enough or is the 'default' section still needed?